pkg/services/hivelocity/device/device.go

```diff
@@ -328,19 +353,58 @@ func (s *Service) actionVerifyShutdown(ctx context.Context) actionResult {
 	// remove condition to reset the timer - we set the condition anyway again
 	conditions.Delete(s.scope.HivelocityCluster, infrav1.DeviceProvisioningSucceededCondition)
 
-	// TODO: make shut down API call
-
+	err = s.scope.HVClient.PowerOffDevice(ctx, deviceID)
+	if err != nil {
+		return actionError{err: fmt.Errorf("[actionDeviceProvisioned] PowerOffDevice failed: %w", err)}
+	}
 	conditions.MarkFalse(
 		s.scope.HivelocityCluster,
 		infrav1.DeviceProvisioningSucceededCondition,
 		infrav1.DeviceShutdownCalledReason,
 		clusterv1.ConditionSeverityInfo,
 		"device shut down has been triggered",
 	)
-
 	return actionContinue{delay: 30 * time.Second}
 }
 
+// Set permanent error with an appropriate label,
+// and go back to the state associate to associate with another device via actionGoBack.
+func (s *Service) setReloadingTooLongTag(ctx context.Context, deviceID int32, provisionCondition *clusterv1.Condition) actionResult {
+	device, err := s.scope.HVClient.GetDevice(ctx, deviceID)
+	if err != nil {
+		s.handleRateLimitExceeded(err, "GetDevice")
+		if errors.Is(err, hvclient.ErrDeviceNotFound) {
+
+			conditions.MarkFalse(
+				s.scope.HivelocityMachine,
+				infrav1.DeviceReadyCondition,
+				infrav1.DeviceNotFoundReason,
+				clusterv1.ConditionSeverityError,
+				fmt.Sprintf("device %d not found", device.DeviceId),
+			)
+			record.Warnf(s.scope.HivelocityMachine, "DeviceNotFound", "Hivelocity device not found")
+			s.scope.HivelocityMachine.SetFailure(capierrors.UpdateMachineError, infrav1.FailureMessageDeviceNotFound)
+			return actionComplete{}
+		}
+		return actionError{err: fmt.Errorf("failed to get associated device: %w", err)}
+	}
+	tags := append([]string(nil), device.Tags...)
```

Comment:
we want to remove the tags that the device got through being associated with this machine. Either by going through the state "dissociate" first or by removing them here already.

Reply:
done.